File: PSET1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int get_height(string prompt);

void print_hash(void);
void print_space(void);
void print_newline(void);

int main(void)
{
    int height = get_height("Enter a number between 1 and 8: ");
    int x, y, z;

    for (x = 1; x <= height; x ++)
    {
        for (y = 0 ; y < height - x; y ++)
        {
            printf(" ");
        }
        for (z = 0; z < height - y; z ++)
        {
            printf("#");
        }
        printf("\n");
    }
}

int get_height(string prompt)
{
    int n;
    do
    {
        n = get_int("%s", prompt);
    }
    while (n < 1 || n > 8);
    return n;
}

File: PSET6/mario/less/mario.py
from cs50 import get_int


def main():
    height = get_height("Enter a number between 1-8 ")

    # Create pyramid based on height.
    for i in range(height):
        print(" " * ((height-i) - 1), end="")
        print("#" * (i+1), end="")
        print()


# Prompt user for height number.
def get_height(prompt):
    while True:
        n = get_int(prompt)
        if n > 0 and n <= 8:
            break
    return n


if __name__ == "__main__":
    main()